src/Http/ApiClient.php

```diff
@@ -346,17 +316,53 @@ protected function determineBodyString($body)
     }
 
     /**
-     * @return string[][]
+     * @param ResponseInterface $response
+     *
+     * @return BunqResponseRaw
      */
-    protected function determineDefaultHeaders()
+    private function createBunqResponseRaw($response)
     {
-        return [
-            self::HEADER_CACHE_CONTROL => [self::HEADER_CACHE_CONTROL_DEFAULT],
-            self::HEADER_USER_AGENT => [self::HEADER_USER_AGENT_BUNQ_SDK_DEFAULT],
-            self::HEADER_GEOLOCATION => [self::HEADER_CUSTOM_GEOLOCATION_DEFAULT],
-            self::HEADER_LANGUAGE => [self::HEADER_CUSTOM_LANGUAGE_DEFAULT],
-            self::HEADER_REGION => [self::HEADER_CUSTOM_REGION_DEFAULT],
-            self::HEADER_CLIENT_REQUEST_ID => [uniqid()],
-        ];
+        $headers = [];
+
+        foreach ($response->getHeaders() as $headerKey => $headerValues) {
+            $headers[$headerKey] = join(self::GLUE_HEADER_VALUE, $headerValues);
```

Comment:
was not familiar with this alias, please use implode(). why php introduces aliases is a big mystery ;-)

Reply:
Fixed!